### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

Updates to the README file.
I cannot update all of the text and links myself. I added TODO labels and any help on this PR is very welcome!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,11 @@ module that contains logic which is able to translate this into a form that is s
 for quantum algorithms. The conversion first creates a FermionicOperator which must then be mapped,
 e.g. by a Jordan Wigner mapping, to a qubit operator in readiness for the quantum computation.
 
+TODO: Update the above:
+- do not mention FermionicOperator but rather general second-quantized ops (at least until we have first-quantized ones)
+- introduce problem classes
+- we don't have any physics yet but the above could be written such that this is not all about chemistry
+
```
Comment (pbark) [inline README.md:19]: This TODO does not see relevant any more. You removed all FermionicOperator mentions from what I see and introduced in the code snipper the problem classes.

**Review by pbark — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mrossinek) [reply README.md:19]: This is not true. See e.g. line 12:

> The conversion first creates a FermionicOperator

File: README.md [hunk 2]
```diff
@@ -43,17 +48,21 @@ to be installed separately.
...
 in hierarchical Data Format 5 (HDF5) format representing all the output data from a chemistry driver.
 
 The [HDF5 driver](https://qiskit.org/documentation/stubs/qiskit.chemistry.drivers.HDF5Driver.html#qiskit.chemistry.drivers.HDF5Driver)
 accepts such HDF5 files as input so molecular experiments can be run, albeit on the fixed data
-as stored in the file. As such, if you have some pre-created HDF5 files from created from Qiskit
+as stored in the file. As such, if you have some pre-created HDF5 files created from Qiskit
 Chemistry, you can use these with the HDF5 driver even if you do not install one of the classical
 computation packages listed above.
 
+TODO: the above paragraph could use some clarification. The sentences are unnecessarily complicated.
```
Comment (mrossinek) [inline README.md:60]: Needs to be addressed

File: README.md [hunk 3]
```diff
@@ -43,17 +48,21 @@ to be installed separately.
...
+
+TODO(below): for different what?
```
Comment (mrossinek) [inline README.md:62]: Needs to be addressed

File: README.md [hunk 4]
```diff
@@ -43,17 +48,21 @@ to be installed separately.
...
 A few sample HDF5 files for different are provided in the
 [chemistry folder](https://github.com/Qiskit/qiskit-community-tutorials/tree/master/chemistry)
+TODO: what will happen to these?
```
Comment (mrossinek) [inline README.md:65]: Needs to be addressed

File: README.md [hunk 5]
```diff
@@ -133,8 +155,13 @@ initial layout of qubits in the mapping and the Terra `PassManager` that will ha
 of the circuits. By passing in a backend as is done above it is internally wrapped into a
 `QuantumInstance` and is a convenience when default setting suffice.
 
+In the end, you are given a result object by the VQE which you can analyze further by interpreting it with
+your problem instance.
+
 ### Further examples
 
+TODO: these links may need updating.
```
Comment (mrossinek) [inline README.md:163]: Needs to be addressed

File: README.md [hunk 6]
```diff
@@ -163,6 +190,8 @@ For questions that are more suited for a forum, we use the **Qiskit** tag in [St
 
 ## Next Steps
 
+TODO: these links may need updating.
```
Comment (mrossinek) [inline README.md:193]: Needs to be addressed

**Review by pbark — CHANGES_REQUESTED** [review]
TODOs and links should be addressed after merging all open PRs

**Review by woodsp-ibm — APPROVED** [review]

**Review by pbark — APPROVED** [review]

**Pull request merged**